src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlConnectionBasicTests.cs

```diff
@@ -23,6 +25,27 @@ public void ConnectionTest()
             connection.Open();
         }
 
+        [Fact]
+        public void ConnectionTestWithCultureTH()
+        {
+            // Save current cultures
+            CultureInfo currentCulture = Thread.CurrentThread.CurrentCulture;
+            CultureInfo currentUICulture = Thread.CurrentThread.CurrentUICulture;
+
+            Thread.CurrentThread.CurrentCulture = new CultureInfo("th-TH");
+            Thread.CurrentThread.CurrentUICulture = new CultureInfo("th-TH");
+
+            using TestTdsServer server = TestTdsServer.StartTestServer();
+            using SqlConnection connection = new SqlConnection(server.ConnectionString);
+            connection.Open();
+
+            // Restore saved cultures
+            Thread.CurrentThread.CurrentCulture = currentCulture;
+            Thread.CurrentThread.CurrentUICulture = currentUICulture;
+
+            Assert.Equal(ConnectionState.Open, connection.State);
```

Comment:
Can you put the restore of culture into a finally block so that if something else (infrastructure failure for example) goes wrong in this test the thread is correctly restored and doesn't have the ability to cause other tests run on the thread to fail.